sherpa/image/DS9.py

```diff
@@ -333,8 +333,16 @@ def xpaset(cmd, data=None, dataFunc=None, template=_DefTemplate,
         stderr=subprocess.STDOUT,
     )
     try:
-        if isinstance(data, six.string_types):
+        # Python 2 vs 3 requires some complexity here.
+        try:  # Python 2 bytes (which are actually strings)
+            unicode(data, "ascii")  # unicode does not exist in Python 3
             data = bytearray(data, "UTF-8")
+        except:
```

Comment:
I agree with catching `NameError`, that's what I usually do (but not in this case, apparently). However, this is just a less ugly (and maybe less explicit, so not even better) way of expressing a
````
if six.PY3:

else:
````

block.